frontend/src/pages/NewRun.tsx

```diff
@@ -322,6 +331,34 @@ class NewRun extends Page<{}, NewRunState> {
         await this.showPageError(`Error: failed to retrieve original run: ${originalRunId}.`, err);
         logger.error(`Failed to retrieve original run: ${originalRunId}`, err);
       }
+    } else if(embeddedPipelineRunId) {
+      try {
+        const runWithEmbeddedPipeline = await Apis.runServiceApi.getRun(embeddedPipelineRunId);
+        const embeddedPipelineSpec = RunUtils.getPipelineSpec(runWithEmbeddedPipeline.run);
+        if (embeddedPipelineSpec) {
+          try {
+            const pipeline = JSON.parse(embeddedPipelineSpec);
+            this.setState({
+              pipeline,
+              pipelineFromRun: pipeline,
+              pipelineName: (pipeline && pipeline.name) || '',
+              usePipelineFromRun: true,
+              usePipelineFromRunLabel: 'Use pipeline from previous step',
```

Comment:
Probably `run` is better than `step` here...  Also "previous" seems a bit confusing.

Alternatively, we could have a pre-filled pipeline selector form field with a button to replace it with another pipeline.

I'm not sure the scenario where the user clones the run, but wants to change the pipeline, is useful. In that case nothing will be taken from  the clone.

Reply:
`page` might make more sense than either. This text covers the case where a user goes to the pipeline details page of a pipeline that only exists inside of a run (it's not uploaded), so saying `run` doesn't really make sense.

This scenario is not about cloning, it's about creating a new run with a pipeline preselected. I agree that switching the pipeline after preselecting it or cloning is an strange path, but I don't want to make restricting assumptions without hearing @ajayalfred's take